ikarus/io/vtkwriter.hh

```diff
@@ -0,0 +1,182 @@
+
+  template <typename RF, Impl::Concepts::DataType DT>
+  void addResultFunction(RF&& resultFunction, DT /*type */) {
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<RF>(resultFunction));
+    else
+      writer_.addPointData(std::forward<RF>(resultFunction));
+  }
+
+  template <template <typename, int, int> class RT, Impl::Concepts::DataType DT>
+  requires(Concepts::ResultType<RT>)
+  void addResult(DT type) {
+    auto resFunction = makeResultVtkFunction<RT>(assembler_);
+    addResultFunction(std::move(resFunction), type);
+  }
+
+  template <Impl::Concepts::DataType DT>
+  void addAllResults(DT type) {
+    using ResultTuple = typename FEType::SupportedResultTypes;
+
+    Dune::Hybrid::forEach(ResultTuple(), [&]<typename RT>(RT i) { addResult<RT::template Rebind>(type); });
+  }
+
+  template <typename GF, Impl::Concepts::DataType DT>
+  void addGridFunction(GF&& gridFunction, const std::string& name, size_t size, DT /* type */) {
+    auto fieldInfo = Dune::Vtk::FieldInfo(name, size);
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<GF>(gridFunction), fieldInfo);
+    else
+      writer_.addPointData(std::forward<GF>(gridFunction), fieldInfo);
+  }
+
+  template <typename GF, typename FieldInfo, Impl::Concepts::DataType DT>
+  requires(std::same_as<FieldInfo, Dune::VTK::FieldInfo> or std::same_as<FieldInfo, Dune::Vtk::FieldInfo>)
+  void addGridFunction(GF&& gridFunction, FieldInfo fieldInfo, DT /* type */) {
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<GF>(gridFunction), fieldInfo);
+    else
+      writer_.addPointData(std::forward<GF>(gridFunction), fieldInfo);
+  }
+
+  template <int dim, typename R, typename Basis, Impl::Concepts::DataType DT>
+  void addInterpolation(R&& vals, const Basis& basis, const std::string& name, DT type) {
+    auto gridFunction = Dune::Functions::makeDiscreteGlobalBasisFunction<Dune::FieldVector<double, dim>>(basis, std::forward<R>(vals));
+    addGridFunction(std::move(gridFunction), name, dim, type);
+  }
+
+  // Other write functions are not explicitly exposed but can be access through writer(). ...
+  std::string write(const std::string& fn, std::optional<std::string> dir = {}) const { return writer_.write(fn, dir); }
+
+  // Legacy functions to add Cell and Point Data
+  template <class... Args>
+  void addCellData(Args&&... args) {
+    writer_.addCellData(std::forward<Args>(args)...);
+  }
+
+  template <class... Args>
+  void addPointData(Args&&... args) {
+    writer_.addPointData(std::forward<Args>(args)...);
+  }
+
+private:
+  const FEContainer& finiteElements() { return assembler_->finiteElements(); }
+  const FERequirement& requirement() { return assembler_->requirement(); }
+
+  UnderlyingVTKWriter writer_;
+  std::shared_ptr<Assembler> assembler_;
+};
+
+// Helpers
+inline auto asCellData() { return Impl::AsCellData{}; }
```

Comment:
Instead of doing these in combination with above, you can also directly create the variable at the struct definition as in [Link](https://github.com/ikarus-project/dune-localfefunctions/blob/1ca4d366b9db2dd5211860525387f35366e162fb/dune/localfefunctions/meta.hh#L32-L33).

See also https://en.cppreference.com/w/c/language/struct_initialization.

Then you really have variables that can be passed to the function and you don't need to add `()` or `{}` at callsite.